### ISSUE TITLE
Proposal for Actor and JLPCActor

### ISSUE THREAD

**Issue by laforge49** [laforge49/JActor#30]
We may be able to simplify things...

public interface Actor {
    Mailbox mailbox();
    Actor parent();
}

public class JLPCActor extends Actor {
    public final Mailbox mailbox;
    public final Actor parent;

```
public Mailbox mailbox() {
    return mailbox;
}

public Actor parent() {
    return parent;
}

public JLPCActor(Mailbox mailbox, Actor parent) {
    this.mailbox = mailbox;
    this.parent = parent;
}
```

}

**Comment by etorreborre** [comment]
Quick question: what happens if an Actor has no parent? Why don't you model this as: `public final Option<Actor> parent`?

**Comment by laforge49** [comment]
Option takes an attribute set, hmm?

Never did really "get" Option in scala, and I think that was more
appropriate. --b

On Sun, Jun 3, 2012 at 6:07 PM, etorreborre <
reply@reply.github.com

**Comment by etorreborre** [comment]
I think that Options are a very good way to avoid NPEs in Java. There are various posts explaining how to implement them in Java, including [this one](http://eng.wealthfront.com/2010/05/better-option-for-java.html).

**Comment by laforge49** [comment]
Looking at that reference, I do like the iterator interface on Option. Not sure that it is the most performant approach.

**Comment by laforge49** [comment]
Whoops! Left off the actor's factory reference.

I still really dislike constructor parameters. Adds a bit of boilerplate to every subclass. Perhaps we should go the other direction and have an initialize method with mailbox, parent and factory arguments. :-)

**Comment by laforge49** [comment]
LPCActor now has a null constructor and an initialize method. --will be released in JActor 3.2.0.
